### ISSUE TITLE
Loading cached DataSilo even if max_seq_len has changed

### ISSUE THREAD

**Issue by tholor** [deepset-ai/FARM#222]
**Describe the bug**
We introduced caching of the DataSilo to speed up preprocessing. However, the checksum of the cache is currently only based on the filename of the training data. This is not enough, since users might change preprocessing params in their script (e.g. max_seq_len) and end up with loading an "outdated" cache  

**Expected behavior**
Don't load the old cache if preprocessing params have changed

**Suggested solution**
Add the params to the checksum here: 
https://github.com/deepset-ai/FARM/blob/master/farm/data_handler/data_silo.py#L221
It must include: `max_seq_len`, `do_lowercase` and `pretrained_model_name_or_path ` (from tokenizer)
It should include: all serializable attributes from the processor ( `metric`, `label_name`, `label_list`, `label_column_name`, `dev_split` ....)
